cli/lib/kontena/cli/grids/health_command.rb

```diff
@@ -0,0 +1,69 @@
+require_relative 'common'
+require "kontena/cli/helpers/health_helper"
+
+module Kontena::Cli::Grids
+  class HealthCommand < Kontena::Command
+    include Kontena::Cli::Common
+    include Kontena::Cli::Helpers::HealthHelper
+    include Common
+
+    parameter "[NAME]", "Grid name"
+
+    def execute
+      require_api_url
+
+      grid = get_grid(name)
+      grid_nodes = client(require_token).get("grids/#{grid['name']}/nodes")
+
+      return show_grid_health(grid, grid_nodes['nodes'])
+    end
+
+    # Validate grid/nodes configuration for grid operation
+    #
+    # @return [Boolean] false if unhealthy
+    def show_grid_health(grid, nodes)
+      initial_size = grid['initial_size']
+      minimum_size = grid['initial_size'] / 2 + 1 # a majority is required for etcd quorum
+
+      grid_health = grid_health(grid, nodes)
+      initial_nodes = nodes.select{|node| node['initial_member']}
+      online_nodes = initial_nodes.select{|node| node['connected']}
+
+      # configuration and status
+      if initial_nodes.length == 0
```

Comment:
Well, each of the conditions could be a separate method that would return or set the error message and do nothing or return true when the validation passes and then you would call them in order.

Like:
```ruby
def validate_foobar
  return true if nodes.foo > bars
  add_error "Foo needs to be more than bars"
  false
end
```